Marlin/src/gcode/bedlevel/abl/G29.cpp

```diff
@@ -285,8 +285,8 @@ G29_TYPE GcodeSuite::G29() {
           G29_RETURN(false);
         }
 
-        const float rx = RAW_X_POSITION(parser.linearval('X', NAN)),
-                    ry = RAW_Y_POSITION(parser.linearval('Y', NAN));
```

Comment:
Removing these modifiers changes the meaning of the parameters and breaks compatibility with previous versions of Marlin. For the most part, coordinates passed to Marlin are relative to the workspace, so that behavior was intentionally preserved here. If you want different behavior to fit some feature, it would be best to add a new flag to get that behavior, or change the way the feature works so that it doesn't depend on changing the meaning of these parameters. Maybe what you really want is to enable `NO_WORKSPACE_OFFSETS` on machines where the existence of workspace offsets gives undesirable results.

Reply:
What we are after here is for probing to follow the workspace offsets. Removing them goes contrary to what we are attempting to do. As it stands the offset applied by M206 was ignored, so when a head was changed it caused the probe to go off the bed. This was especially magnified by the nozzle as probe functionality used by Lulzbot, piezeo sensors such as PyroBall and piezeoelectric, strain gauges like creality and dagoma. If a home offset is applied, we expect everything to follow it.

Reply:
> What we are after here is for probing to follow the workspace offsets.

When you have a workspace offset, whether from `M206` or `G92`, movement commands *should* all be going to different positions than they do without the workspace / home offsets. If any commands or parameters are failing to take the workspace offsets into account, that should be fixed first.

The `RAW_*_POSITION` macros (i.e., `LOGICAL_TO_NATIVE`) are used to convert the coordinate parameters given in relation to the current workspace into coordinates in the native machine space. While it may make sense to treat all coordinates given to "calibration" commands like `G30` as native machine space and eschew workspace offsets, it could be potentially-breaking, so a change like that has to be justified by necessity.

Note that when you're using (`G92`) workspace offsets and you want a command to treat all coordinates in terms of the native machine space, you need to insert a `G53` on the front of the command. Then the command will be run in the native machine space.

For example, if you switched to the third workspace with `G56` and later you wanted to probe the middle of your 200x200 bed, you would need to send `G53 G30 X100 Y100`.

---

But, are you sure that `M206` home offsets are the right tool to apply for your current needs? On IDEX you set the second carriage home position with `M218` and I'm not clear what you are also using `M206` for.